src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3297,6 +3297,7 @@ ImmutableArray<PropertySymbol> addProperties(ImmutableArray<ParameterSymbol> rec
                         {
                             // Deconstruct() is specified to simply assign from this property to the corresponding out parameter.
                             existingOrAddedMembers.Add(prop);
+                            diagnostics.Add(ErrorCode.WRN_UnusedRecordCtorParameter, param.Locations[0], param);
```

Comment:
I don't think that's correct. The parameter can be used even when a user-defined property exists. See linked issue for such a scenario.

Reply:
@jcouv Should I try to somehow surround this by an if condition? or this is not the proper place for this check?

Reply:
@Youssef1313 I'm not sure how to do this correctly and I think it won't be so simple.
We would need to analyze the body with flow analysis. But the analysis of what variables are read is not usually done as part of compilation process, it is usually driven by the IDE calling `CSharpSemanticModel.AnalyzeDataFlow` (which rests on `CSharpDataFlowAnalysis`). But we can't use that because those APIs require syntax to run on, which we don't have in this scenario...

I would probably table this warning for now (doesn't seem worthwhile gain vs. cost at the moment).

To answer your question, I think the initializers are added to the constructor here: https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs#L1028

Reply:
Is the complexity of the fix is because you want to check for user-defined constructors too?
My intent here was to fix that for the SynthesizedRecordConstructor for positional records only.
Since this constructor is auto-generated, I think it must be known during compilation which arguments are being used and which are not. There is something I must be missing.

Reply:
@jcouv isn't it true that record positional parameters are only accessible from initializers? Can we take advantage of that somehow?

Reply:
The trouble is that initializers can use arbitrary expressions. To figure out whether they read or write certain parameters, we still need to do an analysis of those bound nodes.
It can be done but is fairly involved. I'm not sure we want to add an analysis pass for this one warning, or whether an existing pass could be leveraged.

Reply:
Thinking about this more...

Here's how would try doing this:
1. make sure that we're doing a definite assignment pass on the constructor+initializers body (using `DefiniteAssignmentPass.Analyze`)
2. looking at `_usedVariables` (which should be set by `DefiniteAssignmentPass.NoteRead/NoteWrite`) we should be able to determine which primary constructor parameters were unused

Reply:
@Youssef1313 From discussion with the compiler dev team, we feel this would be better addressed by extending the existing IDE analyzer that flags unused parameters and variables. This is not very different from other unused parameters in constructors. Let's see if that approach would be possible. I'm hoping it will be simpler.

Reply:
Sounds reasonable. Given that this PR only touches the compiler-area, I'll close and note in the issue that the fix should be IDE-side.